src/traces/contour/plot.js

```diff
@@ -291,18 +311,303 @@ function makeLines(plotgroup, pathinfo, contours) {
         })
         .style('stroke-miterlimit', 1)
         .style('vector-effect', 'non-scaling-stroke');
+
+    var clipId = clipLinesForLabels ? 'clipline' + cd0.trace.uid : null;
+
+    var lineClip = defs.select('.clips').selectAll('#' + clipId)
+        .data(clipLinesForLabels ? [0] : []);
+    lineClip.exit().remove();
+
+    lineClip.enter().append('clipPath')
+        .classed('contourlineclip', true)
+        .attr('id', clipId);
+
+    Drawing.setClipUrl(lineContainer, clipId);
+
+    var labelGroup = plotgroup.selectAll('g.contourlabels')
+        .data(showLabels ? [0] : []);
+
+    labelGroup.exit().remove();
+
+    labelGroup.enter().append('g')
+        .classed('contourlabels', true);
+
+    if(showLabels) {
+        var labelClipPathData = straightClosedPath(perimeter);
+
+        var labelData = [];
+
+        // invalidate the getTextLocation cache in case paths changed
+        Lib.clearLocationCache();
+
+        var contourFormat;
+        if(contours.labelformat) {
+            contourFormat = d3.format(contours.labelformat);
+        }
+        else {
+            var formatAxis;
+            if(cd0.t.cb) {
+                formatAxis = cd0.t.cb.axis;
+            }
+            else {
+                formatAxis = {
+                    type: 'linear',
+                    _separators: '.,',
+                    _id: 'ycontour',
+                    nticks: pathinfo.length,
+                    showexponent: 'all',
+                    range: [contours.start, contours.end]
+                };
+                setConvert(formatAxis, gd._fullLayout);
+                Axes.calcTicks(formatAxis);
+                formatAxis._tmin = null;
+                formatAxis._tmax = null;
+            }
+            contourFormat = function(v) {
+                return Axes.tickText(formatAxis, v).text;
+            };
+        }
+
+        var dummyText = defs.append('text')
```

Comment:
good question! Actually, I bet if it's already in `Drawing.tester` we can avoid the [`cloneNode` and `appendChild`] steps 🐎